app/grandchallenge/algorithms/templates/algorithms/job_list.html

```diff
@@ -97,8 +115,19 @@ <h5 class="modal-title"></h5>
             let button = $(event.relatedTarget);
             let modal = $(this);
             modal.find('.modal-title').text(button.data('title'));
-            modal.find('.modal-job-output').text(button.data('output'));
+            modal.find('.modal-job-output').html(unescape(button.data('output')));
```

Comment:
Is this safe? I think escaping here might introduce an XSS vulnerability. My thinking: user template -> unsafe code is escaped by `md2html`, which calls `clean` and marked safe as dangerous stuff is escaped -> on L29 _everything_ is escaped including the naughty stuff -> this line unescapes everything including the bad stuff.

Is there another way to pass this information to the modal rather than escaping and unescaping? A javascript string literal maybe? Or by using djangos `json_script` filter https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#json-script (create a unique id for each row and then pass that to the model data insead).

Reply:
The rendered_result_text is text that has been escaped by md2html. It is therefor marked as safe, which is why I need to force_escape (I think), as escape doesn't seem to have any effect. The force_escape/unescape is only needed to escape quotes that are added by the markdown, and mess up the html.

Reply:
Ok - but it also escapes stuff that shouldn't be escapes, introducing the XSS vunrulbility. To see this, add `%3Cscript%3Ealert%28%27uh-oh%27%29%3C/script%3E` to the algorithm template, you should see a pop up when the modal is activated.

Reply:
ah, I see. I'll find another way to do this then.

Reply:
Changed to json_script